### ISSUE TITLE
checkForNode8 is not a function

### ISSUE THREAD

**Issue by brightsider** [firebase/firebase-tools#2944]
### [REQUIRED] Environment info

**firebase-tools:** 9.0.0

**Platform:** macOS Big Sur

### [REQUIRED] Test case

Deploy firebase functions

### [REQUIRED] Steps to reproduce

firebase deploy --only functions

### [REQUIRED] Expected behavior

deploy completed without errors

### [REQUIRED] Actual behavior

✔  functions[testProcessManual(us-central1)]: Successful update operation. 

Error: An unexpected error has occurred.

end of firebase-debug.log:
```
[debug] [2020-12-16T07:32:37.804Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Wed, 16 Dec 2020 07:32:37 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[info] ✔  functions[testProcessManual(us-central1)]: Successful update operation.  
[debug] [2020-12-16T07:32:37.806Z] TypeError: checkForNode8 is not a function
    at /usr/local/lib/node_modules/firebase-tools/lib/deploy/functions/release.js:413:13
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
[error] 
[error] Error: An unexpected error has occurred.
```

**Comment by muhammad-usman-anwar** [comment]
### having Same issue

### [REQUIRED] Environment info
firebase-tools: 9.0.0

Platform: Windows

**Comment by DavuluriHemanthChowdary** [comment]
Same issue I've been facing since yesterday and firebasetools:- 9.0.0 is the version

**Comment by samtstern** [comment]
@brightsider thank you for reporting this, I'll investigate it now.

**Comment by DavuluriHemanthChowdary** [comment]
[debug] [2020-12-16T10:21:21.415Z] ----------------------------------------------------------------------
[debug] [2020-12-16T10:21:21.417Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\heman\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only functions
[debug] [2020-12-16T10:21:21.417Z] CLI Version:   9.0.0
[debug] [2020-12-16T10:21:21.417Z] Platform:      win32
[debug] [2020-12-16T10:21:21.417Z] Node Version:  v15.4.0
[debug] [2020-12-16T10:21:21.419Z] Time:          Wed Dec 16 2020 15:51:21 GMT+0530 (India Standard Time)
[debug] [2020-12-16T10:21:21.419Z] ----------------------------------------------------------------------

**Comment by samtstern** [comment]
The fix is here:
https://github.com/firebase/firebase-tools/pull/2945

We should hopefully be able to get `9.0.1` out today with a fix (once the San Francisco folks wake up!)

**Comment by DavuluriHemanthChowdary** [comment]
> The fix is here:
> #2945
> 
> We should hopefully be able to get `9.0.1` out today with a fix (once the San Francisco folks wake up!)

Nice work! Sir

**Comment by worldofchris** [comment]
Have same issue

**Comment by jwoyo** [comment]
Aaargh! Our CI crashed because it executed "npm install firebase-tools" and then "firebase deploy" – doing so we updated to firebase-tools 9.x automatically and it crashed.

Thank you for reporting & fixing this! ❤️ 
This issue helped me a lot locating the problem.  

We will use a fixed version in our deployment script now. 😅

**Comment by andrewpolemeni** [comment]
I'm having the same issue:
[debug] [2020-12-16T16:33:42.384Z] TypeError: checkForNode8 is not a function
    at C:\Users\Andrew\AppData\Roaming\npm\node_modules\firebase-tools\lib\deploy\functions\release.js:413:13
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
[error] 
[error] Error: An unexpected error has occurred.

**Comment by samtstern** [comment]
The fix has been merged and will be included in the *next*  release (which should be very soon).

**Comment by sharkdevs** [comment]
facing the same issue

**Comment by bkendall** [comment]
San Francisco here: release is on the way out the door. Thanks for your patience! (Downgrading to v8.20 would be a fine workaround)

**Comment by rajprashanthr** [comment]
Thanks a lot

**Comment by bkendall** [comment]
This has shipped in `v9.0.1`!
